streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredVersionedKeyValueStore.java

```diff
@@ -198,6 +280,19 @@ protected void initStoreSerde(final StateStoreContext context) {
                 prepareValueSerde(plainValueSerde, new SerdeGetter(context))
             );
         }
+
+        private byte[] serializeAsBytes(final VersionedRecord<byte[]> versionedRecord) {
+            if (versionedRecord == null) {
+                return null;
+            }
+            final Serde<ValueAndTimestamp<byte[]>> VALUE_AND_TIMESTAMP_SERDE
+                = new ValueAndTimestampSerde<>(new ByteArraySerde());
+            final Serializer<ValueAndTimestamp<byte[]>> VALUE_AND_TIMESTAMP_SERIALIZER
```

Comment:
we should use `try-with-resource` for the serializer to avoid warning about "unclosed" resources